src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -82,15 +87,27 @@ public void put(Collection<SinkRecord> records) {
             logger.debug("SinkTask put - Num records: " + records.size());
 
             Lists.partition(new ArrayList<>(records), this.maxBufferLimit).forEach(batch -> {
-                this.executor.execute(new BatchProcessingThread(batch, this.ablyClient));
+                this.executor.execute(new BatchProcessingThread(batch, this.ablyClient, this.offsetRegistryService));
             });
         }
     }
 
+    /**
+     * precommit is called in regular intervals by the kafka connect
+     * framework. We store that the offsets that are successfully processed
+     * in the offsetRegistryService. This is used to commit the offsets
+     * precommit automatically calls flush.
+     * @param offsets
+     * @return
+     * @throws RetriableException
+     */
     @Override
-    public void flush(Map<TopicPartition, OffsetAndMetadata> map) {
-        // Currently irrelevant because the put call is synchronous
-        return;
+    public Map<TopicPartition, OffsetAndMetadata> preCommit(
+            Map<TopicPartition, OffsetAndMetadata> offsets) throws RetriableException {
+
+        // The offsets map will contain offsets of other topics and partitions
```

Comment:
Is this comment still true? Didn't you say that you confirmed you're actually not getting topic partitions from other workers now?